### ISSUE TITLE
Vaapi not working with webrtc

### ISSUE THREAD

**Issue by EmrysMyrddin** [pion/mediadevices#438]
### Your environment.
  - Version: v0.3.10
  - Browser: none

### What did you do?
Tried to use the VAAPI binding. Seems like broken, but i don't have any clue on how to fix it.

Here are the logs when trying to publish the track with webrtc (using VP8, but it's not working either with VP9) :
```2022/08/16 14:03:12 "msg"="could not set remote description" "error"="video/VP8: video/VP8: libva entrypoint not found\n\nvideo/H264: video/VP8: %!s(<nil>)\n\nvideo/H264: video/VP8: %!s(<nil>)\n\nvideo/H264: video/VP8: %!s(<nil>)"```

**Comment by ZeoWorks** [comment]
VAAPI not working for me either

**Comment by at-wat** [comment]
`libva entrypoint not found` usually means the video card doesn't support specified profile.
You can list supported profiles by `vainfo` command and `VAEntrypointEncSlice` for desired codec is needed.

**Comment by ZeoWorks** [comment]
> `libva entrypoint not found` usually means the video card doesn't support specified profile. You can list supported profiles by `vainfo` command and `VAEntrypointEncSlice` for desired codec is needed.

Hi, both VP8 & VP9 encoding is supported on my machine.
However VAAPI is simply not working on webrtc.
![Screenshot from 2022-10-13 23-01-04](https://user-images.githubusercontent.com/70698824/195719018-6a53a251-668d-408d-823a-6a48d2a4a02b.png)

**Comment by at-wat** [comment]
`VAEntrypointVLD` is for decoding.
`VAEntrypointEncSlice` is required for encoding.

**Comment by ZeoWorks** [comment]
> `VAEntrypointVLD` is for decoding. `VAEntrypointEncSlice` is required for encoding.

That's what I thought also.
I'm really surprised the RX 6700 XT does NOT support VPX hardware encoding.
It's a fairly new & expensive card.

**Comment by at-wat** [comment]
> Will H264/265 encoding be supported in the future with VAAPI?

VAAPI library supports H264/265 and your hardware supports both. (EncSlice for H264 and HEVC)
Need to add wrappers to this package.

**Comment by ZeoWorks** [comment]
> > Will H264/265 encoding be supported in the future with VAAPI?
> 
> VAAPI library supports H264/265 and your hardware supports both. (EncSlice for H264 and HEVC) Need to add wrappers to this package.

Any eta when H264/265 vaapi support?

**Comment by at-wat** [comment]
> Any eta when H264/265 vaapi support?

No, as this is an open source project, contributions are always welcomed.
